src/services/email.integration-test.ts

```diff
@@ -0,0 +1,96 @@
+import EventEmitter from "node:events";
+import nodemailer from "nodemailer";
+import { BotEventNames, TransportConfigurationMissingPayload, TransportReadyPayload } from "../bot-events";
+import { Env, env } from "../utils/env";
+import { expect, getMails, mocks, resolveOnEvent } from "../utils/tests-setup";
+import {
+  ConfigurableSendDeps,
+  EmailMessage,
+  TRANSPORT_NAME,
+  composeMessage,
+  configurableInitialize,
+  configurableSend,
+  notify,
+} from "./email";
+
+describe("Email service", () => {
+  let emitter: EventEmitter;
+
+  beforeEach(() => (emitter = new EventEmitter()));
+  describe("initialize", () => {
+    const fn = configurableInitialize;
+
+    it("should emit an event when it is properly initialized", async () => {
+      const promise = resolveOnEvent(BotEventNames.TRANSPORT_READY, emitter);
+      fn({
+        emitter,
+        env,
+        createTransportFn: nodemailer.createTransport.bind(nodemailer),
+      });
+      const event = (await promise)[0] as TransportReadyPayload;
+      expect(event.name).to.equal(TRANSPORT_NAME);
+    });
+
+    describe("should emit an event when a configuration environment variable missing", () => {
+      const requiredFields: Array<keyof Env> = [
+        "SMTP_HOST",
+        "SMTP_PORT",
+        "SMTP_USER",
+        "SMTP_PASSWORD",
+        "SMTP_FROM",
+        "SMTP_TO",
+      ];
+      requiredFields.forEach((key) => {
+        it(`should emit a missing event when ${String(key)} is missing`, async () => {
+          const promise = resolveOnEvent(BotEventNames.TRANSPORT_CONFIGURATION_MISSING, emitter);
+          const fakeEnv = { ...env, [key]: undefined };
+          fn({
+            emitter,
+            createTransportFn: nodemailer.createTransport.bind(nodemailer),
+            env: fakeEnv,
+          });
+          const event = (await promise)[0] as TransportConfigurationMissingPayload;
+          expect(event.name).to.equal(TRANSPORT_NAME);
+          expect(event.missing).to.have.length(1);
+          expect(event.missing[0]).to.equal(key);
+        });
+      });
+    });
+  });
+
+  describe("notify", () => {
+    const fn = notify;
+
+    it("should notify when the integration is configured", async () => {
```

Comment:
What is this for?

Reply:
Will this only run at startup? Or at multiple times through the app's lifecycle?

Reply:
Sorry, my previous answer was wrong.

The test is checking that once a notification event is requested to be sent via email, it is sent indeed. The title is misleading because it is referencing to things being properly configured when that is not part of the assertions nor the function responsabilities.

I am currently working on a PR that branches from this one, related with sending email to multiple recipients. I'll fix it there.

Good catch, @wadader !